SONARJAVA-5293 Modify rule S6856 to also cover opposite case (#4619)
SonarSource/rspec: rules/S6856/java/rule.adoc

If a method has a path template containing a placeholder, like "/api/resource/{id}", and there's no `@PathVariable` annotation on a method parameter to capture the id path variable, Spring will disregard the id variable.

> This rule will raise an issue if a method does not have corresponding `@PathVariable` annotations and path templates placeholders.

Reviewer: ```suggestion
This rule will raise an issue if a method has a path template with a placeholder, but no corresponding `@PathVariable`, or vice-versa.
```
